pkg/policy/mapstate.go

```diff
@@ -729,15 +927,15 @@ func (keys MapState) DetermineAllowLocalhostIngress() {
 					labels.NewLabel(LabelKeyPolicyDerivedFrom, LabelAllowRemoteHostIngress, labels.LabelSourceReserved),
 				},
 			}
-			es := NewMapStateEntry(nil, derivedFrom, false, isHostDenied, AuthTypeDisabled)
-			keys.DenyPreferredInsert(localRemoteNodeKey, es, nil)
+			es := NewMapStateEntry(nil, derivedFrom, false, isHostDenied, ExplicitAuthType, AuthTypeDisabled) // Authentication never required for remote node ingress
+			keys.denyPreferredInsert(localRemoteNodeKey, es, allFeatures, nil)
 		}
 	}
 }
 
 // AllowAllIdentities translates all identities in selectorCache to their
 // corresponding Keys in the specified direction (ingress, egress) which allows
-// all at L3, without requiring authentication.
+// all at L3, WITHOUT REQUIRING AUTHENTICATION.
```

Comment:
nit: was this capitalization deliberate?

Reply:
Yes, but will revert if you think it is too much of an attention grab. I had to reason about the auth requirements on every call site and having this pop out makes it more obvious why the combo `ExplicitAuthType, AuthTypeDisabled` is used.

Reply:
It certainly grabbed my attention :sweat_smile: 

What wasn't clear to me is the implications here. I'd prefer a more verbose explanation, even with scary words like "Warning", or "Be aware" etc. that help the reader to reason about this rather than just seeing the capital letters and being left to figure that part out :-)